### PR TITLE
[Datasets] Update `ExecutionPlan` repr formatting to use consistent spacing

### PR DESCRIPTION
**Pull request by scottjlee**
## Why are these changes needed?
- Use a default indenting of three spaces `"   "` in the custom formatting logic used in `ExecutionPlan.get_plan_as_string()` to maintain consistency with existing `trailing_space` logic
- Updating remaining docs according to the changes in https://github.com/ray-project/ray/pull/32722, which are currently breaking the CI docs tests.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/data/_internal/plan.py [hunk 1]
```diff
@@ -270,11 +271,11 @@ def get_plan_as_string(self) -> str:
                         )
                     schema_str.append(f"{col_str}")
                 schema_str = ",\n".join(schema_str)
-                schema_str = "{\n" + schema_str + "\n\t}"
+                schema_str = "{\n" + schema_str + f"\n{INDENT_STR}}}"
             dataset_str = (
-                "Dataset(\n\tnum_blocks={},\n\tnum_rows={},\n\tschema={}\n)".format(
-                    num_blocks, count, schema_str
-                )
+                f"Dataset(\n{INDENT_STR}num_blocks={num_blocks},"
```
Comment (c21) [inline python/ray/data/_internal/plan.py:276]: Can we follow the other logic in this method, to take `trailing_space` into consideration? We should also add trailing spaces as there might be more operators.

example:

```
ds = ray.data.read_csv("/Users/chengsu/ray/python/ray/data/examples/data/iris.csv")
ds = ds.map_batches(lambda x:x)
ds = ds.map_batches(lambda x:x)
ds = ds.map_batches(lambda x:x)
ds = ds.map_batches(lambda x:x)
ds = ds.map_batches(lambda x:x)
ds

MapBatches(<lambda>)
+- MapBatches(<lambda>)
   +- MapBatches(<lambda>)
      +- MapBatches(<lambda>)
         +- MapBatches(<lambda>)
            +- Dataset(
    num_blocks=1,
    num_rows=150,
    schema={
        sepal.length: double,
        sepal.width: double,
        petal.length: double,
        petal.width: double,
        variety: string
    }
)

```

File: doc/source/data/getting-started.rst [hunk 2]
```diff
@@ -62,7 +62,17 @@ transform datasets. Ray executes transformations in parallel for performance at
 .. testoutput::
 
     MapBatches(transform_batch)
-    +- Dataset(num_blocks=1, num_rows=150, schema={sepal length (cm): double, sepal width (cm): double, petal length (cm): double, petal width (cm): double, target: int64})
+    +- Dataset(
+        num_blocks=1,
```
Comment (bveeramani) [inline doc/source/data/getting-started.rst:66]: Is `num_blocks` deterministic? If not, might be less flaky to use ellipsis here

```
.. testoutput::
    :options: +ELLIPSIS

    +- Dataset(
        num_blocks=...,
        num_rows=150
```

File: python/ray/data/tests/test_dataset_numpy.py [hunk 3]
```diff
@@ -122,8 +122,8 @@ def test_numpy_roundtrip(ray_start_regular_shared, fs, data_path):
     ds.write_numpy(data_path, filesystem=fs)
     ds = ray.data.read_numpy(data_path, filesystem=fs)
     assert str(ds) == (
-        "Dataset(\n\tnum_blocks=2,\n\tnum_rows=?,"
-        "\n\tschema={__value__: ArrowTensorType(shape=(1,), dtype=int64)}\n)"
+        "Dataset(\n    num_blocks=2,\n    num_rows=?,"
+        "\n    schema={__value__: ArrowTensorType(shape=(1,), dtype=int64)}\n)"
```
Comment (bveeramani) [inline python/ray/data/tests/test_dataset_numpy.py:126]: Could do this later, but we probably shouldn't assert against `repr`s. There's a high maintenance cost. `repr`s can change between versions, and we'd have to update the tests each time.

Comment (scottjlee) [reply python/ray/data/tests/test_dataset_numpy.py:126]: would the best alternative be to directly inspect the `ds` object's attributes, e.g. `ds.num_blocks == 2`, etc?

Comment (bveeramani) [reply python/ray/data/tests/test_dataset_numpy.py:126]: Not sure if we'd want to test `num_blocks` for this test in particular, but yeah, I think directly inspecting makes sense. Like, we could test the schema with `Dataset.schema` and the data values with `Dataset.takeall` or something.

Comment (bveeramani) [reply python/ray/data/tests/test_dataset_numpy.py:126]: Anyway, doesn't need to block this PR -- just something to think about.

**Review by c21 — APPROVED** [review]
LGTM, thanks @scottjlee!

**Comment by scottjlee** [comment]
The failing `Documentation` test [error](https://buildkite.com/ray-project/oss-ci-build-pr/builds/14162#0186c842-5f88-4918-8e6e-68b5f22d958e/1831-2236) looks unrelated:
`/ray/doc/source/ray-air/examples/index.rst:29: WARNING: unknown document: /ray-air/examples/serving`

**Review by clarkzinzow — APPROVED** [review]

**Pull request merged**